cgs_vmc/layers.py

```diff
@@ -405,3 +406,38 @@ def _build(self, inputs: tf.Tensor) -> tf.Tensor:
 
     batch_components = [self._mps_var[:, :, index] for index in index_inputs]
     return tf.stack(batch_components, axis=0)
+
+
+class GraphConvLayer(snt.AbstractModule):
+  """GraphConvLayer module with adjacency list"""
+
+  def __init__(
+      self,
+      output_channels: int,
+      adj: np.ndarray,
+      name: str = 'gnn_layer'):
+    """Constructs GraphConvLayer moduel.
+
+    Args:
+      output_channels: Number of output channels in convolution.
+      adj: Adjacency list of the graph that stores indices of neighbors and itself for every site, with shape [n_site, num_neighbor].
+      name: Name of the module.
+    """
+    super(GraphConvLayer, self).__init__(name=name)
+    self._output_channels = output_channels
+    self._adj = adj
+
+  def _build(self, inputs: tf.Tensor) -> tf.Tensor:
+    """Connects the GraphConvLayer module into the graph, with input `inputs`.
+
+    Args:
+      inputs: Tensor with input values.
+
+    Returns:
+      Result of convolving `inputs` with adjacency matrix and module's kernels.
+    """
+    num_neighbor = np.shape(adj)[1]
```

Comment:
This actually doesn't work, as `adj` is not defined in this scope. Use the class attribute `self._adj`.